File: sample-java/src/main/java/social/bigbone/sample/GetTagTimeline.java
package social.bigbone.sample;

import social.bigbone.MastodonClient;
import social.bigbone.api.Pageable;
import social.bigbone.api.entity.Status;
import social.bigbone.api.exception.BigBoneClientInstantiationException;
import social.bigbone.api.exception.BigBoneRequestException;

import static social.bigbone.api.method.TimelineMethods.StatusOrigin.LOCAL_AND_REMOTE;

public class GetTagTimeline {
    public static void main(final String[] args) throws BigBoneRequestException, BigBoneClientInstantiationException {
        final String instance = args[0];
        final String hashtag = args[1];

        // Instantiate client
        final MastodonClient client = new MastodonClient.Builder(instance).build();

        // Get statuses from public timeline
        final Pageable<Status> statuses = client.timelines().getTagTimeline(hashtag, LOCAL_AND_REMOTE).execute();
        statuses.getPart().forEach(status -> System.out.println(status.getContent()));
    }
}

File: sample-kotlin/src/main/kotlin/social/bigbone/sample/GetTagTimeline.kt
package social.bigbone.sample

import social.bigbone.MastodonClient
import social.bigbone.api.Range
import social.bigbone.api.method.TimelineMethods

object GetTagTimeline {
    @JvmStatic
    fun main(args: Array<String>) {
        val instance = args[0]
        val hashtag = args[1]

        // Instantiate client
        val client = MastodonClient.Builder(instance)
            .build()

        // Get statuses from public timeline
        val statuses = client.timelines.getTagTimeline(hashtag, Range(), TimelineMethods.StatusOrigin.LOCAL_AND_REMOTE).execute()
        statuses.part.forEach {
            println(it.content)
        }
    }
}
